types/src/system/auction/detail.rs

```diff
@@ -205,10 +205,34 @@ pub(crate) fn process_unbond_requests<P: Auction + ?Sized>(provider: &mut P) ->
             // current era id + unbonding delay is equal or greater than the `era_of_creation` that
             // was calculated on `unbond` attempt.
             if current_era_id >= unbonding_purse.era_of_creation() + unbonding_delay {
-                // Move funds from bid purse to unbonding purse
-                provider
-                    .unbond(unbonding_purse)
-                    .map_err(|_| Error::TransferToUnbondingPurse)?;
+                match unbonding_purse.new_validator_public_key() {
+                    Some(public_key) => {
+                        let bid = provider
+                            .read_bid(&public_key.to_account_hash())?
+                            .ok_or(Error::ValidatorNotFound)?;
```

Comment:
We need to be absolutely certain that this can't fail outside of some local data corruption or similar.  Same for the `provider.handle_delegation()` call immediately after this.

Seems like if any of these calls return errors we should log an error and continue.  If the `read_bid` call returns `Ok(None)` we should probably just unbond.

Currently if an invalid validator is specified in the `new_validator_public_key`, this will cause the `process_unbond_requests` to return an error which would cause none of the unbonding requests to be actioned, and the whole `commit_step()` to fail.

Reply:
> Ok(None) we should probably just unbond.

correct

Reply:
@darthsiroftardis fix this as soon as you can, thx

Reply:
The fix is present and a corresponding test has been added